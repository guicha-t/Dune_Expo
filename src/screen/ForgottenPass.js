import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Text, StyleSheet, TouchableOpacity, Image} from 'react-native';
import { observer } from 'mobx-react';

import Store from './../global/store/Store'

@observer
export default class ForgottenPass extends Component {
  navigationOptions: {
  drawerLockMode: 'locked-closed'
}
  constructor(props) {
    super(props);

  this.state = {
      username: '',
      password: '',
    };
  }

  onLogin() {
    const { username, password } = this.state;

    fetch('http://176.31.252.134:9001/api/v1/login/reset', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: username,
      }),
    }).then((response) => response.json())
        .then((responseJson) => {
          if (responseJson.status == 200) {
	    this.props.navigation.navigate('Start')
            Alert.alert('E-mail de récupération envoyé')
          }
          else {
            Alert.alert('Adresse E-mail introuvable')
          }
        })
        .catch((error) => {
          console.error(error);
        });
  }

  render() {
    return (
      <View style={styles.container}>
	<View style={{flex:0.4, alignItems: 'center', justifyContent:'center',}}>
	<Text style={{fontSize:20,}}>
	Réinitialiser votre mot de passe
	</Text>
	</View>
        <View style={{flex:0.1, alignItems: 'center', justifyContent:'center'}}>
          <TextInput
            value={this.state.username}
            onChangeText={(username) => this.setState({ username })}
            placeholder={'Adresse e-mail'}
            autoCapitalize = 'none'
            style={styles.input}
          />
          <Button
            title={'Envoyer'}
            style={styles.input}
            color='#363453'
            onPress={this.onLogin.bind(this)}
          />
      </View>
      <View style={{flex: 0.1, alignItems: 'center', justifyContent: 'flex-end', marginBottom: 20}}>
        <TouchableOpacity onPress={()=>this.props.navigation.navigate('Start')}>
          <Text>
            Retour
          </Text>
        </TouchableOpacity>
      </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FEE599',
  },
  input: {
    width: 200,
    height: 44,
    padding: 10,
    borderWidth: 0,
    borderColor: 'black',
    marginBottom: 10,
  },
});
