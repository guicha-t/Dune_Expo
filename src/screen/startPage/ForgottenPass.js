import React, { Component } from 'react';
import { Alert, TextInput, View, Text, StyleSheet, ToastAndroid, TouchableOpacity, KeyboardAvoidingView, Image} from 'react-native';
import { observer } from 'mobx-react';

import { Button, Icon } from 'react-native-elements';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import { Fumi } from 'react-native-textinput-effects';
import AlertPro from "react-native-alert-pro";

import Store from './../../global/store/Store'
import * as cfg from "./../../Config";

@observer
export default class ForgottenPass extends Component {

static navigationOptions = {
   drawerLockMode: 'locked-open',
}

  constructor(props) {
    super(props);

  this.state = {
      username: '',
      password: '',
    };
  }

  onLogin() {
    const { username, password } = this.state;

    fetch(cfg.API_URL + '/login/reset', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: username,
      }),
    }).then((response) => response.json())
        .then((responseJson) => {
          if (responseJson.status == 200) {
              this.props.navigation.navigate('Start')
              ToastAndroid.showWithGravityAndOffset(
                'Email envoyé avec succès.',
                ToastAndroid.LONG,
                ToastAndroid.BOTTOM,
                25,
                50,
              );
          }
          else {
            this.AlertPro.open()
          }
        })
        .catch((error) => {
          console.error(error);
        });
  }

  render() {
    return (
      <View style={styles.container}>
        <KeyboardAvoidingView style={styles.container} behavior="padding">
          <View style={{flex: 0.5}}>
            <Image
              style={{flex: 1, height: undefined, width: undefined}}
              source={require('./../../picture/header/dunelogo.png')}
              resizeMode="contain"
              />
          </View>

           <View style={{flex:0.4}}>
            <View style={{flex: 0.2, justifyContent:'center', alignItems:'center'}}>
              <Text style={styles.primetextblue}>RECUPERATION DU MOT DE PASSE</Text>
            </View>
            <View style={{flex: 0.6, justifyContent:'center', alignItems:'center'}}>
              <Fumi
                label={'E-mail'}
                style={{ width: 220, backgroundColor:'#363453'}}
                value={this.state.username}
                onChangeText={(username) => this.setState({ username })}
                iconClass={FontAwesomeIcon}
                iconName={'at'}
                iconColor={'#FFF'}
                labelStyle={{ color: '#FFF' }}
                iconSize={20}
                iconWidth={40}
                inputPadding={16}
                />
            </View>

            <View style={{flex: 0.2,flexDirection:'row'}}>
              <View style={{flex: 0.5, flexDirection:'row', justifyContent:'center', alignItems:'center'}}>

                <Button
                  title=""
                  onPress={()=>this.props.navigation.navigate('Start')}
                  icon={{
                   type: 'font-awesome',
                   name: 'arrow-left',
                   size: 15,
                   color: 'white',
                 }}
                  buttonStyle={{
                    backgroundColor: '#363453',
                    borderWidth: 2,
                    borderColor: 'white',
                    borderRadius: 30,
                    width: 60,
                    alignItems:'center',
                    paddingLeft: 20,
                  }}
                  containerStyle={{ marginVertical: 10, height: 50, width: 250 }}
                  titleStyle={{ fontWeight: 'bold' }}
                />

              </View>

              <View style={{flex: 0.5, flexDirection:'row', justifyContent:'center', alignItems:'center'}}>

                <Button
                  title=""
                  onPress={this.onLogin.bind(this)}
                  icon={{
                   type: 'font-awesome',
                   name: 'check',
                   size: 15,
                   color: 'white',
                 }}
                  buttonStyle={{
                    backgroundColor: '#363453',
                    borderWidth: 2,
                    borderColor: 'white',
                    borderRadius: 30,
                    width: 60,
                    alignItems:'center',
                    paddingLeft: 20,
                  }}
                  containerStyle={{ marginVertical: 10, height: 50, width: 250 }}
                  titleStyle={{ fontWeight: 'bold' }}
                />

              </View>
            </View>
          </View>
          <View style={{flex: 0.1}}></View>
        </KeyboardAvoidingView>

        <AlertPro
           ref={ref => {
             this.AlertPro = ref;
           }}
           onConfirm={() => this.AlertPro.close()}
           showCancel={false}
           title="OUPS !"
           message="L'email est introuvable"
           textConfirm="REESAYER"
           closeOnPressMask={true}
           customStyles={{
             mask: {
               backgroundColor: "transparent"
             },
             container: {
               borderWidth: 1,
               borderColor: "#ea4335",
               shadowColor: "#000000",
               shadowOpacity: 0.1,
               shadowRadius: 10
             },
             buttonConfirm: {
               backgroundColor: "#ea4335"
             }
           }}
         />

      </View>
        );
      }
    }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FEE599',
  },
  input: {
    width: 220,
    height: 44,
    padding: 10,
    borderWidth: 1,
    borderColor: '#363453',
  },
  primetextblue: {
    fontSize: 20,
    color: '#363453',
    fontWeight: 'bold'
  },

});
