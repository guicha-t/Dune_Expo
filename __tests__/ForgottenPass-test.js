import ForgottenPass from './../src/screen/startPage/ForgottenPass';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render ForgottenPass correctly', () => {

    const tree = renderer.create(<ForgottenPass />).toJSON();
    expect(tree).toMatchSnapshot();
});