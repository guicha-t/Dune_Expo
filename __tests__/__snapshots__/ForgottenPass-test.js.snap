// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ForgottenPass correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f6c6d2",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#f6c6d2",
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.5,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2A5E85",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          RECUPERATION DU MOT DE PASSE
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.6,
            "justifyContent": "center",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "overflow": "hidden",
                "paddingTop": 16,
              },
              Object {
                "backgroundColor": "#2A5E85",
                "width": 220,
              },
              Object {
                "height": 64,
              },
            ]
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#f0f0f0",
                  "position": "absolute",
                  "top": 8,
                  "width": 1,
                },
                Object {
                  "height": 48,
                  "left": 48,
                },
              ]
            }
          />
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 48,
                "left": 56,
                "position": "absolute",
                "top": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              E-mail
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            animationDuration={300}
            height={48}
            iconClass={[Function]}
            iconColor="#FFF"
            iconName="at"
            iconSize={20}
            iconWidth={40}
            inputPadding={16}
            label="E-mail"
            labelStyle={
              Object {
                "color": "#FFF",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            passiveIconColor="#a3a3a3"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "padding": 7,
                  "paddingLeft": 0,
                },
                Object {
                  "color": "#FFF",
                  "marginLeft": 56,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 0.2,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ea4335",
                      "borderColor": "white",
                      "borderRadius": 30,
                      "borderWidth": 2,
                      "paddingLeft": 20,
                      "width": 60,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#4caf50",
                      "borderColor": "white",
                      "borderRadius": 30,
                      "borderWidth": 2,
                      "paddingLeft": 20,
                      "width": 60,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text />
                <Text
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.1,
        }
      }
    />
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderColor": "#ea4335",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginHorizontal": 30,
            "maxWidth": 350,
            "shadowColor": "#000000",
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 25,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              OUPS !
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#666",
                    "fontSize": 17,
                    "paddingTop": 10,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              L'email est introuvable
            </Text>
          </View>
          <View
            style={
              Object {
                "borderRadius": 5,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ea4335",
                  "borderRadius": 5,
                  "marginBottom": 20,
                  "marginHorizontal": 10,
                  "minWidth": 90,
                  "opacity": 1,
                  "paddingVertical": 8,
                }
              }
              testID="buttonConfirm"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontSize": 15,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                REESAYER
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
